RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractRecycleBin.java

```diff
@@ -0,0 +1,552 @@
+            
+            List<AbstractFile> rFiles = rFileMap.get(rFileName);
+            
+            if(rFiles == null) {
+                return;
+            }
+            SleuthkitCase skCase = Case.getCurrentCase().getSleuthkitCase();
+            for (AbstractFile rFile : rFiles) {
+                if (context.dataSourceIngestIsCancelled()) {
+                    return;
+                }
+
+                if (iFile.getParentPath().equals(rFile.getParentPath()) 
+                        && iFile.getMetaFlagsAsString().equals(rFile.getMetaFlagsAsString())) {
+                    try {
+                        postArtifact(createArtifact(rFile, recycleBinArtifactType, metaData.getFileName(), userName, metaData.getDeletedTimeStamp()));
+                        if(rFile instanceof FsContent) {
+                            if(rFile.isDir()) {
+                                AbstractFile directory = getOrMakeFolder(Case.getCurrentCase().getSleuthkitCase(), (FsContent)rFile, metaData.getFileName());
+                                popuplateDirectory(Case.getCurrentCase().getSleuthkitCase(), directory, rFile.getChildren(), metaData.getFileName(), metaData.getDeletedTimeStamp());
+                                
+                            } else {
+                                AbstractFile folder = getOrMakeFolder(Case.getCurrentCase().getSleuthkitCase(), (FsContent)rFile.getParent(), Paths.get(metaData.getFileName()).getParent().toString());
+                                addFileToDirectory(skCase, rFile, folder, Paths.get(metaData.getFileName()).getFileName().toString(), metaData.getDeletedTimeStamp());
+                            }
+                        }
+                    } catch (TskCoreException ex) {
+                        logger.log(Level.WARNING, String.format("Unable to add attributes to artifact %s", rFile.getName()), ex); //NON-NLS
+                    }
+                }
+            }
+        } finally {
+            (new File(tempFilePath)).delete();
+        }
+    }
+    
+    /**
+     * 
+     * @param skCase
+     * @param folder
+     * @param children
+     * @param parentPath
+     * @param deletedTimeStamp
+     * @throws TskCoreException 
+     */
+    private void popuplateDirectory(SleuthkitCase skCase, AbstractFile folder, List<Content> recycledChildren, String parentPath, long deletedTimeStamp) throws TskCoreException {
+        if (recycledChildren == null) {
+            return;
+        }
+        
+        for(Content child: recycledChildren) {
+            if(child instanceof AbstractFile) {
+                AbstractFile fileChild = (AbstractFile) child;
+                if(fileChild.isFile()) {
+                    addFileToDirectory(skCase, fileChild, folder, fileChild.getName(), deletedTimeStamp);
+                } else if(fileChild.isDir()) {
+                    String newPath = parentPath + File.separator + fileChild.getName();
```

Comment:
You should probably just hard code in "/" since that is what the DB uses and not rely on Java's, which I think is platform dependent.

Reply:
Now that you say that it should actually be the "\" because that path is going to be a windows absolute path from the $I file.  getOrMake will take care of correcting the slashes.